app/controllers/concerns/pagination.rb

```diff
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+module Pagination
+  def max_page(products_count)
+    (products_count.to_f / ApplicationController::PER_PAGE).ceil
+  end
+
+  def correct_page(page)
+    if page < 1
+      1
+    elsif page > max_page(Product.count)
```

Comment:
if you will provide product count from the controller, you will make this methods more generic and reusable for other entities, and will remove 2 queries to DB from concerns